int16[] data 		#[orientation, index]